apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -166,39 +166,57 @@ public override string ConnectionString
         public Task<SpannerTransaction> BeginReadOnlyTransactionAsync(TimestampBound targetReadTimeStamp,
             CancellationToken cancellationToken = default(CancellationToken))
         {
+            if (targetReadTimeStamp.Mode == TimestampBoundMode.MinReadTimestamp
+                || targetReadTimeStamp.Mode == TimestampBoundMode.MaxStaleness)
+            {
+                throw new ArgumentException(nameof(targetReadTimeStamp),
+                    "MinReadTimestamp and MaxStaleness can only be used in a single-use"
```

Comment:
Use interpolated strings and nameof(...) expressions here in case the enum member names change.